src/Psalm/Internal/Analyzer/ClassAnalyzer.php

```diff
@@ -973,15 +974,17 @@ public function analyze(
             $file_manipulations = [];
 
             foreach ($plugin_classes as $plugin_fq_class_name) {
-                if ($plugin_fq_class_name::afterStatementAnalysis(
+                $event = new AfterClassLikeAnalysisEvent(
```

Comment:
Do we need to reinitialize the event for every hook call?

Reply:
We don't have control of what the plugin will do with the event. They could unset properties for example. Also file_manipulation is risky to keep.

Reply:
which makes me think I should merge file_manipulations and push a clean array each time I think. I'll do that this afternoon